weldx/util.py

```diff
@@ -1205,15 +1205,34 @@ class _Eq_compare_nested:
 
     @staticmethod
     def _compare(x, y):
-        if type(x) is not type(y):
-            return False
+        # treat special cases first
+        # rely upon the fact, that comparison would fail, if y is of an incompatible type.
```

Comment:
I think the question here is if we want to test for equality (meaning in types as well) or "just" compare compatibilty with `__eq__`?
Why not test for type equality?
I would prefer having the security for cases where it would matter (ASDF file roundtrips)

Reply:
Type equality is not very pythonic. We already encountered the case of the ndarray wrapper, for which we would have to create an ugly workaround. How does exact type matching increase security and what are ASDF file roundtrips?

Reply:
If I save a dictionary into an ASDF file and read it back into a new dict it would be good to know everything gets deserialized into the original type (in some cases)
personally I would like to rely on the method to test for this as well and would prefer to catch a few known exceptions over having some ambiguity

Reply:
So the only case here (so far) is the asdf ndarray wrapping, am I right?

Reply:
yes not aware of anything else so far

Reply:
I've added a list of tuples to handle future cases of enforced type equalities.